src/Test.FunctionalTests.Sarif/MultitoolCommandLineTests.cs

```diff
@@ -35,5 +36,64 @@ public void Multitool_LaunchesAndRunsSuccessfully()
                 process.ExitCode.Should().Be(0);
             }
         }
+
+#if DEBUG
```

Comment:
We need to test the analyze command hang, in sdk we only have analyze-test which is also wrapped under `#if DEBUG` so my test will also need to be wrapped under it.
I have tested ` .\BuildAndTest.cmd -Configuration Debug` will all pass.

Reply:
I was confused.  Shaopeng educated me offline.  Issue is we fill the channel without yielding our thread.  So you really do need 10k+ files, because that's the channel size.

Reply:
Also: did you go back and verify that the UT hangs on the unmodified code? I think that's important for this tes.
----- yes, undo the fix and leave test can repro and have the error in the screenshot in the other comment.

as talk in IM we need 10241 files.

Will work on the change you recommended in IM.